### PR TITLE
[ci] Implement automatic tests on GH Actions on linux, windows and darwin

### PR DESCRIPTION
**Pull request by Lomanic**
Based off https://github.com/mvdan/github-actions-golang/blob/3710bfcc3e71fc692118d0d5a42df57481095db7/.github/workflows/test.yml

Linked to #734

### CODE REVIEW

**Comment by Lomanic** [comment]
Maybe we should remove circleci tests, they really want me to create an account on their platform just to see logs for failing jobs, which is really annoying, and they are now just redundant with these Github Actions.

File: v3/process/process_darwin.go [hunk 1]
```diff
@@ -212,17 +210,18 @@ func (p *Process) GidsWithContext(ctx context.Context) ([]int32, error) {
 }
 
 func (p *Process) GroupsWithContext(ctx context.Context) ([]int32, error) {
-	k, err := p.getKProc()
-	if err != nil {
-		return nil, err
-	}
+	return nil, common.ErrNotImplementedError
```
Comment (shirou) [inline v3/process/process_darwin.go:213]: Sorry, I can not remember but is this `Groups` not working?

**Review by shirou — COMMENTED** [review]
It looks awesome! Thank you so much! I commented to just one method. if ok, let's merge and remove CircleCI!

File: v3/process/process_darwin.go (hunk 1 above)
Comment (Lomanic) [reply v3/process/process_darwin.go:213]: I don't think it ever worked, see #913.

Comment (shirou) [reply v3/process/process_darwin.go:213]: thanks!

**Review by shirou — APPROVED** [review]
LGTM!

**Pull request merged**